pkg/registry/servicecatalog/instance/strategy.go

```diff
@@ -153,23 +153,19 @@ func (instanceRESTStrategy) PrepareForUpdate(ctx genericapirequest.Context, new,
 	newServiceInstance.Spec.ServiceClassRef = oldServiceInstance.Spec.ServiceClassRef
 	newServiceInstance.Spec.ServicePlanRef = oldServiceInstance.Spec.ServicePlanRef
 
-	// TODO: We currently don't handle any changes to the spec in the
-	// reconciler. Once we do that, this check needs to be removed and
-	// proper validation of allowed changes needs to be implemented in
-	// ValidateUpdate. Also, the check for whether the generation needs
-	// to be updated needs to be un-commented.
-	newServiceInstance.Spec = oldServiceInstance.Spec
+	// Clear out the ServicePlanRef so that it is resolved during reconciliation
+	if newServiceInstance.Spec.ExternalServicePlanName != oldServiceInstance.Spec.ExternalServicePlanName {
+		newServiceInstance.Spec.ServicePlanRef = nil
+	}
 
 	// Spec updates bump the generation so that we can distinguish between
 	// spec changes and other changes to the object.
-	//
-	// Note that since we do not currently handle any changes to the spec,
-	// the generation will never be incremented
 	if !apiequality.Semantic.DeepEqual(oldServiceInstance.Spec, newServiceInstance.Spec) {
 		if utilfeature.DefaultFeatureGate.Enabled(scfeatures.OriginatingIdentity) {
 			setServiceInstanceUserInfo(newServiceInstance, ctx)
 		}
 		newServiceInstance.Generation = oldServiceInstance.Generation + 1
+		setServiceInstanceReadyFalseCondition(newServiceInstance)
```

Comment:
In a follow-up, we need to take this out.  I'm sorry that I missed it earlier, but update should _not_ affect the status of a resource. It's okay that the ready condition remains set after this - the controller will eventually unset it. Remember, the system is not transactional and only has to eventually converge on the right state.